set.mm

```diff
@@ -658423,6 +658453,374 @@ base set if and only if the neighborhoods (convergents) of every point
+      ( cvv wcel crg cmnring co clmod elexd wi cif wceq eleq1d eqid eleq1 dedth
+      c0 oveq2 imbi2d ccnfld oveq1 cbs cfv cfrlm cnring elimhyp 0ex mnringlmodi
+      elimel sylc ) ADGZHZBIZHZBDJZKZLZHZADCFMEUPURVBNURBUPDUAZOZUSKZVAHZNDVCDV
+      DPZVBVFURVGUTVEVADVDBUSUBQUCURVFURBUDZOZVDUSKZVAHBVHBVIPVEVJVABVIVDUSUEQV
+      DUFUGZVIUOVDVIVKUHKZVKRVLRURVIUQHVHUQHBVHBVIUQSVHVIUQSUIUJDVCUOUKUMULTTUN
+      $.
+  $}
+
+  ${
+    $d ph x y $.  $d .+ x y $.  $d .xb x y $.  $d .0b x y $.  $d A x y $.
+    $d B x y $.  $d R a b i x y $.  $d M a b i x y $.  $d V a b i x y $.
+    $d X a b i x y $.  $d Y a b i x y $.
+    mnringmulrvald.1 $e |- .xb = ( .r ` R ) $.
+    mnringmulrvald.2 $e |- .0b = ( 0g ` R ) $.
+    mnringmulrvald.3 $e |- A = ( Base ` M ) $.
+    mnringmulrvald.4 $e |- .+ = ( +g ` M ) $.
+    mnringmulrvald.5 $e |- V = ( R freeLMod A ) $.
+    mnringmulrvald.6 $e |- B = ( Base ` V ) $.
+    mnringmulrvald.7 $e |- .x. = ( .r ` ( R MndRing M ) ) $.
+    mnringmulrvald.8 $e |- ( ph -> R e. U ) $.
+    mnringmulrvald.9 $e |- ( ph -> M e. W ) $.
+    mnringmulrvald.10 $e |- ( ph -> X e. B ) $.
+    mnringmulrvald.11 $e |- ( ph -> Y e. B ) $.
+    $( Value of multiplication in a monoid ring.  (Contributed by Rohan
+       Ridenour, 13-May-2024.) $)
+    mnringmulrvald $p |- ( ph -> ( X .x. Y ) =
+      ( V gsum ( a e. A , b e. A |->
+        ( i e. A |-> if (
+          i = ( a .+ b ) ,
+          ( ( X ` a ) .xb ( Y ` b ) ) ,
+          .0b
+        ) )
+      ) ) ) $=
+      ( vx vy cv co wceq cfv cif cmpt cmpo cgsu cvv cmnring mnringmulrd syl6eqr
+      cmulr eqcomd fveq1 oveqan12d ifeq1d mpteq2dv mpoeq3dv oveq2d adantl ovexd
+      wa ovmpod ) AUIUJMNCCKPQBBIBIUKPUKZQUKZDULUMZVOUIUKZUNZVPUJUKZUNZFULZOUOZ
+      UPZUQZURZULZKPQBBIBVQVOMUNZVPNUNZFULZOUOZUPZUQZWFULZGUSAUIUJCCWGUQZGAWOEJ
+      UTULVCUNGAUIUJBCDEFHIJKLOPQRSTUAUBUCUEUFVAUDVBVDVRMUMZVTNUMZVMZWGWNUMAWRW
+      EWMKWFWRPQBBWDWLWRIBWCWKWRVQWBWJOWPWQVSWHWAWIFVOVRMVEVPVTNVEVFVGVHVIVJVKU
+      GUHAKWMWFVLVN $.
+  $}
+
+  ${
+    $d C c d $.  $d A a b c d $.  $d B a b c d $.
+    mnringmulrcldlem.1 $e |- ( ph -> ( a e. A /\ b e. B ) ) $.
+    mnringmulrcldlem.2 $e |- ( ph -> P = <. a , b >. ) $.
+    mnringmulrcldlem.3 $e |- C e. V $.
+    $( Lemma for ~ mnringmulrcld .  Value of an operation given by maps-to
+       notation.  (Contributed by Rohan Ridenour, 13-May-2024.) $)
+    mnringmulrcldlem $p |- ( ph -> ( ( a e. A , b e. B |-> C ) ` P ) = C ) $=
+      ( vc vd cv cmpo co cfv wa wceq nfcv cop fveq2d df-ov syl6reqr csb nfcsb1v
+      wcel nfcsbw csbeq1a sylan9eq cbvmpo oveqi equcom sylbir eqcomd eqid elexi
+      anbi12i ovmpoa syl5eq syl eqtr3d ) AGNZHNZGHBCDOZPZEVEQZDAVGVCVDUAZVEQVFA
+      EVHVEJUBVCVDVEUCUDAVCBUGVDCUGRZVFDSIVIVFVCVDLMBCHMNZGLNZDUEZUEZOZPDVEVNVC
+      VDGHLMBCDVMLDTMDTGHVJVLGVJTGVKDUFUHHVJVLUFVCVKSZVDVJSZDVLVMGVKDUIHVJVLUIU
+      JZUKULLMVCVDBCVMDVNVKVCSZVJVDSZRZDVMVTVOVPRDVMSVOVRVPVSGLUMHMUMURVQUNUOVN
+      UPDFKUQUSUTVAVB $.
+  $}
```

Comment:
Maybe this could be added/named as a more general theorem?

Reply:
Maybe "ovmpopr" (value for a maps-to operation at a pair).

Reply:
~mnringmulrcldlem is more like [~fvmpt2](https://us.metamath.org/mpeuni/fvmpt2.html) than [~fvmpt](https://us.metamath.org/mpeuni/fvmpt.html), so I'll probably call it ~fvmpopr2d (not ~ov... since it's using function application rather than operation application)